packages/model/src/tests/workspace.test.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Workspace,
+  WorkspaceOptions,
+  ProjectKind,
+  Service,
+  WebApp,
+  Package,
+} from '..';
+import {Base, Options} from '../base';
+
+// Test creation of different types of workspaces
+describe('Workspace', () => {
+  const name = 'test';
+  const root = '../../';
+
+  const options: Options = {
+    name,
+    root,
+  };
+  const base = new Base(options);
+  const webApp = new WebApp(options);
+  const pkg = new Package(options);
+  const service = new Service(options);
+  const webApps: WebApp[] = [webApp];
+  const packages: Package[] = [pkg];
+  const services: Service[] = [service];
+  const emptyWebApps: WebApp[] = [];
+  const emptyPackages: Package[] = [];
+  const emptyServices: Service[] = [];
+
+  it('creates a workspace for a WebApp', () => {
+    const opts: WorkspaceOptions = {
+      webApps,
+      services: emptyServices,
+      packages: emptyPackages,
+      name,
+      root,
+    };
+    const webAppWorkspace = new Workspace(opts);
+    expect(webAppWorkspace.projects[0].kind).toBe(ProjectKind.WebApp);
```

Comment:
Nice one. The `kind` property is what we'll probably want to use as differentiation between all these different `Base` extended classes. New readers can now self discover this with these tests. 👍